src/containers/FlashFirmware/FlashFirmware.tsx

```diff
@@ -89,6 +91,10 @@ export const ManageFlashFirmware = () => {
                     addNotification(notification())
                     return true
                 }
+                if (!hideModal()) {
+                    setOpenModal({ open: true, type: MODAL_TYPE.OPENIRIS })
```

Comment:
My only suggestion would be to improve naming cause right now the modal type explains more or less nothing. How about `MODAL_TYPE.BEFORE_FLASHING`?